### PR TITLE
Reordered Magic Hatching Potions, fixes issue 8465

### PR DESCRIPTION
**Pull request by willira2**
[//]: #8465 
Fixes https://github.com/HabitRPG/habitica/issues/8465

### Changes
[//]: # Updated Magic Hatching Potion list in hatching-potions.js to order the potions in a more logical pattern. Moved Royal Purple potion to top of the list, and other seasonal potions are now listed from Spring to Winter.

[//]: # 18119562

----
UUID:

### CODE REVIEW

File: website/common/script/content/hatching-potions.js [hunk 1]
```diff
@@ -49,44 +49,44 @@ let drops = {
 };
 
 let premium = {
-  Spooky: {
```
Comment (SabreCat) [inline website/common/script/content/hatching-potions.js:52]: Looks like whitespacing got adjusted by accident--it doesn't have a functional impact, but the standard in our code is two spaces per indent.

**Review by SabreCat — CHANGES_REQUESTED** [review]

File: website/common/script/content/hatching-potions.js (hunk 1 above)
Comment (willira2) [reply website/common/script/content/hatching-potions.js:52]: Whoops, I'll fix that. I didn't notice it in Sublime Text but I can see it in vim.

**Comment by SabreCat** [comment]
@willira2: Nice! Of course, now there are Cupid Potions to incorporate. :cat::sweat_drops: If you rebase this branch onto `develop`, that'll resolve the merge conflict. And maybe the Cupid potions could go to the top, preceding the springtime potions.

**Comment by willira2** [comment]
@SabreCat To do this, I just need to go onto the branch where I was working on this and run "git rebase develop"? I'm still learning git D:

**Comment by Alys** [comment]
@willira2 First do `git fetch upstream` to download the latest material from the official repository.

Then do `git rebase upstream/develop` on the branch you were working on.

**Comment by willira2** [comment]
@Alys Thank you! Ok, I've done that. Hopefully that resolves the conflict?

**Comment by Alys** [comment]
Ah, sorry, I should have said: To make the rebase be added to your pull request, you also need to push your changes to your own fork of the repo in GitHub. That will be something like `git push origin issue8465`

**Comment by willira2** [comment]
@SabreCat Do you mind taking me off this issue? Changing the file is no problem, but my git skills (or lack thereof) are proving to be the challenge. It's such an easy fix that I hate to keep it in limbo like this.

**Comment by SabreCat** [comment]
@willira2 All good--I can do the rebase and incorporate the Cupid potions. What's your Habitica UUID so I can credit you for the rest?

**Comment by Alys** [comment]
@willira2 We can take you off the issue if you wish, but we can also talk you through using git. This is a low priority issue so there's no rush to get it done!

If there were merge conflicts when you rebased, we can help you resolve them. If you're not sure, and if the output of the rebase is still on screen, you can paste it here or take a screenshot. 

If you're having trouble pushing the rebased code to github, I can give you more detailed steps.

Just let us know! :)

**Comment by willira2** [comment]
@SabreCat I'll give it one last shot with @Alys 's help. 

@Alys , rebasing went through, but when I went to push, this is what I got:

screen shot 2017-02-03 at 10 56 31 pm

Thanks to both of you for helping me!

**Comment by Alys** [comment]
(@SabreCat sorry! Your reply hadn't appeared when I'd posted mine.)

@willira2 That's not an uncommon thing and is usually easy to fix, but it can occasionally indicate a problem that needs to be resolved first, so before pushing we should check that what you're about to push is okay. Can you please do this:
`git diff origin/issue8465  website/common/script/content/hatching-potions.js`
and show us the output?

And by the way, if it all seems to go horribly wrong, we can recover your work! Git lets us go back to good commits so fixes are easy. :)

**Comment by willira2** [comment]
@SabreCat ok, here's the output of that command:

screen shot 2017-02-03 at 11 18 46 pm

**Comment by Alys** [comment]
@willira2 Can you do this command and show us the output?

`git remote  -v`

It sounds like `origin` might not be the local name for your GitHub fork.

**Comment by willira2** [comment]
@Alys 
screen shot 2017-02-04 at 12 09 34 am

**Comment by Alys** [comment]
That's all good, and now that I look at the error message more closely, it's complaining about the path to the file - sorry about that, I should have noticed.

Were you running `git diff origin/issue8465 website/common/script/content/hatching-potions.js` in the top-level directory in your local install? I.e., in the `habitica` directory that contains files like config.json and package.json? That's where it should be run.

**Comment by willira2** [comment]
@Alys  oh whoops, I was not. This looks better: 
screen shot 2017-02-04 at 12 07 22 pm

**Comment by SabreCat** [comment]
Great! There's a couple of things going on here:

* You can see changes to this file that took place while you were working on this PR in that `git diff` output: `return pot._season === 'winter'` was replaced with `return pot._season === 'valentines'` because we moved from Winter Wonderland to the Valentine's mini event. But git doesn't know if your PR is supposed to switch it back to `winter` or not. Make sure that line includes the `valentines` bit instead in your local branch, before committing and pushing.
* When you rebase, your local branch ends up with a different sequence of commits than the one you have stored on GitHub. That's why you get the `failed to push some refs` error, because git can't resolve the situation by just tacking a new commit on to the end of your PR. To tell git to overwrite the remote branch, making it back into an exact copy of your local branch, use the `--force` switch on your `git push`.

**Comment by willira2** [comment]
@SabreCat Ok, I've checked the file and made sure it was correct, per your specifications. I'm going to try committing and pushing it again with the --force switch.

**Review by SabreCat — APPROVED** [review]

**Comment by SabreCat** [comment]
Perfect! @willira2, what's your Habitica UUID, if any? You'll get contributor tier credit for this!

**Comment by willira2** [comment]
@SabreCat , sorry, just saw this! My UUID is 75e6d5ff-9e9d-4ddf-a03c-bc2e0e07605b. Thanks again to you and @Alys for the help!

**Comment by SabreCat** [comment]
@willira2 Great! Congratulations on becoming a Tier 1 Blacksmith. 😺 Note that subsequent tiers require increasing amounts of effort from one to the next, so Tier 2 will require multiple PRs or a more substantial PR to attain. But keep helping out and we'll express our gratitude accordingly!

**Pull request merged**